app/views/rooms/_room_chat.html.erb

```diff
@@ -1,59 +1,83 @@
-
-
 ///////////////////////////////////////////////////////////////////////////////
 ///////////////////////////////====CHAT====////////////////////////////////////
 ///////////////////////////////////////////////////////////////////////////////
-  var $sendButton = $("#js-send-button-im");
-  var $inputIM    = $("#js-input-im");
-  var $chatBody   = $("#js-chat-body");
-  var $speechRecBody = $("#speech-rec-body");
-  var localId = comm.getLocalID();
-
-  $sendButton.on('click', sendIM);
-  $inputIM.on('keypress', checkEnterKey);
-
-  // With Turbolinks pages will change without a full reload, so you can't rely on the usual window.beforepageunload event. Instead Turbolinks fires events on document to provide hooks into the lifecycle of the page.
-  $(document).on("page:fetch", function(){
-  console.log('turbolinks page fetch event');
-  });
-
-
-  // send a message
-  function sendIM() {
-      "<% current_user.refresh_token_if_expired %>"
-      var content = "<%= current_user.name %>" + "@@@" + "<%= current_user.uid %>" + "@@@" +
-    "<%= current_user.oauth_token %>" + "@@@" + $inputIM.val() ;
-    $inputIM.val("");
+var $sendButton = $("#js-send-button-im");
+var $inputIM = $("#js-input-im");
+var $chatBody = $("#js-chat-body");
+var $speechRecBody = $("#speech-rec-body");
+var localId = comm.getLocalID();
+$sendButton.on('click', sendIM);
+$inputIM.on('keypress', checkEnterKey);
+// With Turbolinks pages will change without a full reload, so you can't rely on the usual window.beforepageunload event. Instead Turbolinks fires events on document to provide hooks into the lifecycle of the page.
+$(document).on("page:fetch", function() {
+    console.log('turbolinks page fetch event');
+});
+// send a message
+function sendIM() {
+    "<% current_user.refresh_token_if_expired %>"
+    var message = $inputIM.val();
+    var content = "<%= current_user.name %>" + "@@@" + "<%= current_user.uid %>" + "@@@" + "<%= current_user.oauth_token %>" + "@@@" + message;
     comm.send(content)
-    // append what you send to your own page
-    appendNewIM({
-        data: content,
-        callerID: "<%= current_user.name %>"
-    })
-  }
-
-  // send your benedict query result to everyone else in the chat
-  function sendBennyReply(reply){
-        "<% current_user.refresh_token_if_expired %>"
-      var content = "benny" + "@@@" + "<%= current_user.name %>" + "@@@" + "<%= current_user.uid %>" + "@@@" +
-    "<%= current_user.oauth_token %>" + "@@@" + reply;
-    comm.send(content)
-    // append what you send to your own page
-    appendNewBennyReply({
-        data: content,
-        callerID: "<%= current_user.name %>"
-    })
-  }
-
-  // click the send button when enter key is pressed
-  function checkEnterKey(event) {
-    if ( event.keyCode == 13 ) {
-      $sendButton.click();
+    // send to redis if you are the user that originally created this message
```

Comment:
this is where we post new json replies from the benny python server to the rails controller to store in redis.